### PR TITLE
Fixing potential integer overflow on sequence counter

### PR DESCRIPTION
**Pull request by shamisp**
Current implementation may potential cause hangs or data corruption

### CODE REVIEW

File: transformer_engine/pytorch/csrc/userbuffers/userbuffers.cu [hunk 1]
```diff
@@ -51,6 +51,10 @@
     asm volatile("fence.sc.gpu;\n");                                                               \
   }
 
+// Return true if producer > consumer, otherwise false while preventing integer overflow
+// If we expect that produce will be 2B+ messages behind consumer
```
Comment (deepakn94) [inline transformer_engine/pytorch/csrc/userbuffers/userbuffers.cu:55]: Nit: "produce" -> "producer"

**Review by deepakn94 — COMMENTED** [review]
LGTM, minor comment.

File: transformer_engine/pytorch/csrc/userbuffers/userbuffers.cu (hunk 1 above)
Comment (shamisp) [reply transformer_engine/pytorch/csrc/userbuffers/userbuffers.cu]: fixed

**Review by ksivaman — APPROVED** [review]
Got the greenlight from Anton.

**Pull request merged**